srtcore/sync.h

```diff
@@ -402,7 +419,9 @@ class CSyncTpl
     /// @retval false Resumed due to being past @a timestamp
     bool wait_for(const steady_clock::duration& delay)
     {
+        THREAD_PAUSED();
         bool signaled = m_cond->wait_for(*m_locker, delay);
+        THREAD_RESUMED();
```

Comment:
`THREAD_PAUSED()` better remove from this place. I think it should be in the upper-level code.

Reply:
Maybe it should be moved to inside `CCondVar::wait_for`, or even better, to the one in `SyncEvent`?

Reply:
I think we should ask @jeandube regarding the preferred way as he uses this macro.
But the main point is that int we better restore the initial usage outside of the waiting functions. Because now we have the following remaining, but `THREAD_PAUSED()` is already removed:
```
void *CUDT::tsbpd(void *param)
{
    CUDT *self = (CUDT *)param;

    THREAD_STATE_INIT("SRT:TsbPd");
```

Reply:
Ok, then I'll remove them. We'll find a better way to restore them later.

Reply:
For THREAD_PAUSE/RESUME, higher the level better the info reported by the thread watchdog and crash log. Regarding THREAD_STATE_INIT the last time I cleanded this was to sync the threads reported by smgr debug and top -H.